### ISSUE TITLE
Constant-value fees

### ISSUE THREAD

**Issue by zawy12** [monero-project/monero#3766]
I'm told Monero is interested in stable-valued fees.  If POW's were not changed, and if Moore's law has mostly died out, there is a good approximate measure of real-world price already on the block chain.  

constant-value-fee = F2 = F1 / D1 * D2 * R1 / R2 * M

Where F1 is the fee you've selected today that is implicitly based on a current external world dollar amount per coin.  F2 is what the transaction fee should at some time in the future.  D and R are difficulty and reward.  M could be some estimate of Moore's law based on times between 1 and 2. 

But don't be a moocher.  Send me a donation if you use this idea.

**Comment by Gingeropolous** [comment]
yeah, i've been talking about this kind of thing for a while. I would posit you need to include the blocksize in the fee... the difficulty is the perceived value of the coin (a lot of people hopping on to mine), whereas the blocksize is the actual value - a lot of people are transacting. Both parameters capture an aspect of value that can be used to serve a proxy for external blockchain value. 

I think the question is figuring out your magic number M.

**Comment by zawy12** [comment]
This is the general equation that relates future coin price to future difficulty:

```P1*(R1 + nb1*Fb1 + total1*Fp1) / D1 * M1 = P2*(R2 + nb2*Fb2 + total2*Fp2) / D2 * M2 ```

Where
1 and 2  = measures at time 1 (now) and time 2 (future)
P = USD price / coin
R = coins/block reward
D = difficulty  ( D = hashes\*T / 2^x where x is leading zeros in maxTarget)
nb and Fb = number of txns per block and base coin fee per txn
total = total coins transferred per block
Fp = Percent fee per coin transferred
M = "Moore's law" hashes per USD mining expense.

If the relative rate of Fb and Fp does not change, Fp = k\*Fb, so you can easily solve for the future Fb2 and Fp2 given a beginning Fb1.  

To show it from a bigger perspective, the above is:

(P=USD/block) / (D=hashes/block) * (M=hashes/USD)  = a unit-less constant ratio

USD here is a substitute for constant value.  The fees in this equation are constant value even if USD value changes because USD cancels.   If the POW can be made to be mostly an electrical expense rather than hardware (see my POW idea), then this constant value metric is based on electricity which is the best single-commodity metric of constant value.  

This can be used to create a constant-value coin, but I haven't written the white paper yet.
